### ISSUE TITLE
False Negative `mandatory-version`  Warning

### ISSUE THREAD

**Issue by volodya-lombrozo** [objectionary/lints#207]
I'm getting the following warning when I check my `xmir`:

```
XMIR is incorrect: [[mandatory-version WARNING]:0 The +version meta is mandatory, but is absent
```

but I have the `version`:

```xml
<program xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         dob="2023-03-19T00:00:00"
         ms="0"
         name="j$MethodByte"
         revision="1234567"
         time="2025-01-09T08:43:15.990916Z"
         version="1.2.3">
...
```

There is how I check the `xmir`:

```java
final Collection < Defect > defects = new Program(new StrictXmir(this.doc)).defects()
    .stream()
    .collect(Collectors.toList());
if (!defects.isEmpty()) {
    throw new IllegalStateException(
        String.format(
            "XMIR is incorrect: %s, %n%s%n",
            defects,
            this.doc
        )
    );
}
```

Full `xmir`:
[MethodByte.xmir.txt](https://github.com/user-attachments/files/18358754/MethodByte.xmir.txt)

Please, fix the check.

**Comment by volodya-lombrozo** [comment]
Blocks: https://github.com/objectionary/jeo-maven-plugin/issues/946

**Comment by volodya-lombrozo** [comment]
@yegor256 Could you have a look, please?

**Comment by yegor256** [comment]
@volodya-lombrozo this warning is not about `/xmir/@version`, but about meta that is supposed to be in the `<metas>`:

```
<xmir>
  <metas>
    <meta>
      <head>+version</head>
      <tail>0.0.0</tail>
    </meta>
  </metas>
</xmir>
```

**Comment by volodya-lombrozo** [comment]
@yegor256 I believe it should be mentioned in the error message, otherwise even if I close this ticket, somebody else will rise it again very soon. What do you think?

**Comment by h1alexbel** [comment]
@yegor256 @volodya-lombrozo besides the error messages, we also have `motive`-s that explain lints and provide examples. I believe we should make them accessible, and link them to produced error messages. WDYT?

**Comment by volodya-lombrozo** [comment]
@h1alexbel Would be nice

**Comment by volodya-lombrozo** [comment]
> @volodya-lombrozo this warning is not about `/xmir/@version`, but about meta that is supposed to be in the `<metas>`:
> 
> ```
> <xmir>
>   <metas>
>     <meta>
>       <head>+version</head>
>       <tail>0.0.0</tail>
>     </meta>
>   </metas>
> </xmir>
> ```

@yegor256 I'm getting this for you code example: 

```
cvc-pattern-valid: Value '+version' is not facet-valid with respect to pattern '[a-z]+[a-z0-9]*' for type '#AnonType_headmeta'.
  cvc-type.3.1.3: The value '+version' of element 'head' is not valid.
```

This is one more prove that highlights the fact that we need proper accessible documentation with examples.

**Comment by yegor256** [comment]
@volodya-lombrozo it's a typo, should be:

```
<xmir>
  <metas>
    <meta>
      <head>version</head>
      <tail>0.0.0</tail>
    </meta>
  </metas>
</xmir>
```

**Comment by h1alexbel** [comment]
@volodya-lombrozo still an issue?

**Comment by volodya-lombrozo** [comment]
@h1alexbel I'll check it soon

**Comment by volodya-lombrozo** [comment]
@h1alexbel The message is still rather cryptic, there are still no lines or context:

```
Caused by: java.lang.IllegalArgumentException: 2 error(s) in XML document: cvc-pattern-valid: Value '+version' is not facet-valid with respect to pattern '[a-z]+[a-z0-9]*' for type '#AnonType_headmeta'.;cvc-type.3.1.3: The value '+version' of element 'head' is not valid.
```
It's just hard to work with such messages. But I hope it will be solved with https://github.com/objectionary/lints/pull/349
